From 1f3c3c808940c5855586f2a195e643f835c6975f Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Tue, 1 Apr 2014 17:06:21 +0800
Subject: [PATCH 3847/5965] PD #89278: fix compile error

this commit fix compil error of commit 8b6b4ee05fc1e4325b076b1d9a7524e41fdc4b65

by shuai.li@amlogic.com & zhe.wang@amlogic.com
---
 sound/soc/aml/m6/aml_audio_codec_probe.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/aml/m6/aml_audio_codec_probe.c b/sound/soc/aml/m6/aml_audio_codec_probe.c
index e7f82f5e6875..9ad3a0536712 100755
--- a/sound/soc/aml/m6/aml_audio_codec_probe.c
+++ b/sound/soc/aml/m6/aml_audio_codec_probe.c
@@ -319,10 +319,10 @@ static int aml_audio_codec_probe(struct platform_device *pdev)
 	struct device_node* child;
 	struct i2c_board_info board_info;
 	struct i2c_adapter *adapter;
+	struct i2c_client *client;
 	aml_audio_codec_info_t temp_audio_codec;
 #ifdef CONFIG_SND_AML_M6TV_AUDIO_CODEC
 	struct tas5711_platform_data *pdata;
-	struct i2c_client *client;
 
 	pdata = kzalloc(sizeof(struct tas5711_platform_data), GFP_KERNEL);
 	if (!pdata) {
@@ -358,8 +358,8 @@ static int aml_audio_codec_probe(struct platform_device *pdev)
 
 static int aml_audio_codec_remove(struct platform_device *pdev)
 {
+#ifdef CONFIG_SND_AML_M6TV_AUDIO_CODEC
 	int i;
-
 	for(i = 0; i < prob_priv.num_eq; i++){
 		if(prob_priv.eq_configs[i].regs)
 			kfree(prob_priv.eq_configs[i].regs);
@@ -370,6 +370,7 @@ static int aml_audio_codec_remove(struct platform_device *pdev)
 
 	if(prob_priv.sub_bq_table)
 		kfree(prob_priv.sub_bq_table);
+#endif
 
     return 0;
 }
-- 
2.19.0

